src/library.js

```diff
@@ -38,6 +38,10 @@ LibraryManager.library = {
     setTempRet0(val);
   },
 
+  $zeroMemory: function(address, size) {
+    HEAPU8.fill(0, address, address + size);
```

Comment:
How about adding something like this:
```js
#if LEGACY_VM_SUPPORT
  if (!HEAPU8.fill) {
    (for var i = 0; i < size; i++) {
      HEAPU8[address + i] = 0;
    }
    return;
  }
#endif
```

Reply:
Good idea!

Reply:
I do kind of worry, in general, about adding complexity like this before knowing if we really need it because it will make it really hard to ever know it is used, and its really hard to test.

Reply:
We can test it, if we think it's worth it - do `HEAPU8.fill = null` to force the "polyfill" to be used, and build with legacy VM support. I think we have some tests like that.

Reply:
Do you want me to go back to add one?  I'm tempted to leave it.